rdcp.c

```diff
@@ -1098,14 +1099,17 @@ static int rdcp_test_client(struct rdcp_cb *cb)
 		    perror("Couldn't open file");
 		    goto out;
 		}
-		cb->fp = fdopen(cb->fd, "rb");
-		fseek(cb->fp, 0, SEEK_END);
-		cb->metadata.size = ftell(cb->fp);
+		// cb->fp = fdopen(cb->fd, "rb");
+		// fseek(cb->fp, 0, SEEK_END);
+		// cb->metadata.size = ftell(cb->fp);
+		stat(cb->metadata.src_path, &filebuf);
+		cb->metadata.size = filebuf.st_size;
```

Comment:
how is using stat() instead of fseek() helps here? stat() is great and has better performance but beside that i dont see how it fix an issue. if you can change your commit to explain to use stat() instead of fseek() its fine by me to accept the commit anyway. but it is requiring some explanation why/how it fixes the issue with small files.

Reply:
I'm inspired by [issue#8](https://github.com/roidayan/rdcp/issues/8). After read the issue, I searched the Internet for the difference between stat() and fseek(). I believed that it might have some conflicts between a file pointer(cb->fp) and a file descriptor(cb->fd), so it'd better use just a file pointer or just a file descriptor rather than hybird, and using stat() can simply and easily get the whole file's length without moving position(which I thought the bug happened), but actually I didn't know exactly why this happened, sorry.

Reply:
ok and it is fine by me but need to refactor the patch to have a meaningful title and description.
you can just say use stat() to get the file size and the body will be to fix issue #8 and its fine by me. also need to clean commented out code. there is no point to leave this.